src/main/java/org/jenkinsci/plugins/workflow/multibranch/AbstractWorkflowBranchProjectFactory.java

```diff
@@ -63,14 +64,19 @@ public abstract class AbstractWorkflowBranchProjectFactory extends BranchProject
 
     @NonNull
     @Override public WorkflowJob setBranch(@NonNull WorkflowJob project, @NonNull Branch branch) {
-        project.setDefinition(createDefinition());
-        BranchJobProperty property = project.getProperty(BranchJobProperty.class);
         try {
-            if (property == null) {
-                project.addProperty(new BranchJobProperty(branch));
-            } else { // TODO may add equality check if https://github.com/jenkinsci/branch-api-plugin/pull/36 or equivalent is implemented
-                property.setBranch(branch);
-                project.save();
+            BulkChange bc = new BulkChange(project);
```

Comment:
As of https://github.com/jenkinsci/jenkins/pull/2428 https://javadoc.jenkins.io/hudson/BulkChange.html suggests using a `try`-with-resources block (where `commit` is _not_ called in a `finally` block) as the normal usage idiom. https://github.com/jenkinsci/jenkins/blob/daa43ec13dd1bc56a56a742ef5e4476313ebd865/core/src/main/java/jenkins/model/Jenkins.java#L1893-L1899 is typical. Is there a particular reason to use a different idiom here?